### ISSUE TITLE
Problems with endpointscfg.py script (ImportError: No module named _endpointscfg_setup)

### ISSUE THREAD

**Issue by bigunyak** [cloudendpoints/endpoints-python#54]
Hi there,

I've discovered a number of problems with the current implementation of `endpointscfg.py`, could you please comment on them and maybe suggest a workaround or a fix.

1. I use buildout to setup my development environment, that means all packages are installed as eggs and scripts with appropriate `sys.path` patching are placed under bin/ directory. Here is what I get when try to run `bin/endpointscfg.py` script:
    ```
    ./bin/endpointscfg.py get_openapi_spec main.EchoApi --hostname echo-api.endpoints.my-test-project.appspot.com
    Traceback (most recent call last):
      File "./bin/endpointscfg.py", line 89, in <module>
        import _endpointscfg_setup  # pylint: disable=unused-import
    ImportError: No module named _endpointscfg_setup
    ```
    Seems that the script has to be ran from it's source directory, which is very inconvenient.
2. OpenAPI configuration generation apparently involves application loading (`main.py`) but `appengine_config.py` file is not executed before that in contrast to when application is ran with `dev_appserver.py` for example. So, all the code to setup paths properly I have in the `appengine_config.py` needs to be duplicated somewhere...
3. The last it would be very nice to be able to run the script and not necessarily be in the application root directory, just point it to the app root directory instead.

**Comment by bigunyak** [comment]
I forgot to mention one more problem.
4. After adding all necessary paths to `PYTHONPATH`, I'm still not able to run the script, what I get is:
```
Traceback (most recent call last):
  File "libs/endpoints/endpointscfg.py", line 636, in <module>
    main(sys.argv)
  File "libs/endpoints/endpointscfg.py", line 632, in main
    args.callback(args)
  File "libs/endpoints/endpointscfg.py", line 490, in _GenOpenApiSpecCallback
    application_path=args.application)
  File "libs/endpoints/endpointscfg.py", line 335, in _GenOpenApiSpec
    application_path=application_path)
  File "libs/endpoints/endpointscfg.py", line 186, in GenApiConfig
    module = __import__(module_name, fromlist=base_service_class_name)
  File "APPROOT/parts/echoapp/main.py", line 3, in <module>
    import endpoints
  File "APPROOT/eggs/google_endpoints-2.0.3-py2.7.egg/endpoints/__init__.py", line 29, in <module>
    from apiserving import *
  File "APPROOT/eggs/google_endpoints-2.0.3-py2.7.egg/endpoints/apiserving.py", line 74, in <module>
    from google.api.control import client as control_client
ImportError: No module named api.control
```
The solution which I found is to import `google` module inside `endpointscfg.py` file. But why is this error, could you please fix it?

**Comment by bigunyak** [comment]
Ah, I found there are `--application` and `--output` command line options for the `endpointscfg.py` script, so please, scrape my third point. :)
The `--service` option is not documented though (or I haven't found where). And it's not clear what should be the value for `--hostname` parameter, in quickstart it's suggested as `echo-api.endpoints.[PROJECT-ID].appspot.com` and in "Adding Required API Management" section it's `your-service.appspot.com` where `your-service` suggested to be your project ID. One would expect these options to be better explained.

**Comment by ubragg** [comment]
I ran into your issue 2 here (endpointscfg.py doesn't run appengine_config.py), and found that issue is being tracked here: https://issuetracker.google.com/issues/35898132

**Comment by inklesspen** [comment]
I believe these issues should all be resolved as of 2.4.3. Please comment if you're still having issues.
